resources/js/src/app/main.js

```diff
@@ -229,43 +234,64 @@ if ( headerParent )
         {
             const vueApp = document.getElementById("vue-app");
 
-            let bodyOffset = 0;
+            vueApp.style.marginTop = headerHeight + "px";
+            vueApp.style.minHeight = "calc(100vh - " + headerHeight + "px)";
+        }
+    }
+
+    // Set descending z-index for all header elements and create list of elements with unfixed class for later use
+    function prepareHeaderElements()
+    {
+        if (headerLoaded && !App.isShopBuilder)
+        {
+            headerParent = headerParent.offsetParent ? headerParent : document.querySelector("[data-header-offset]");
+
+            let zIndex = 100;
 
-            for ( let i = 0; i < headerParent.children.length; i++ )
+            for (let i = 0; i < headerParent.children.length; i++)
             {
-                bodyOffset += headerParent.children[i].getBoundingClientRect().height;
+                const elem = headerParent.children[i];
+
+                elem.style.zIndex = zIndex;
+                zIndex--;
+
+                isUnfixedElement.push(elem.classList.contains("unfixed"));
```

Comment:
Die Prüfung auf `classList.contains()` sollte so performant sein, dass es m.M.n. unnötig ist, diesen Wert extra nochmal in einer Variablen zu speichern. Außerdem bekommst du ein Problem, wenn die Klasse "unfixed" im Nachhinein durch JS entfernt oder hinzugefügt wird.

Reply:
done